clientGQL_test.go

```diff
@@ -66,36 +71,94 @@ func ATestClient(t *testing.T, endpoint string) *ol.Client {
 		autopilot.GraphQLQueryFixtureValidation(t, fmt.Sprintf("%s_request.json", endpoint)))))
 }
 
+func NewTestRequest(request string, variables string, response string) TestRequest {
+	templater := NewTestDataTemplater()
+	testRequest := TestRequest{
+		Request:   templater.ParseTemplatedString(request),
+		Variables: templater.ParseTemplatedString(variables),
+		Response:  templater.ParseTemplatedString(response),
+	}
+	if !IsValidJson(testRequest.Variables) {
+		panic(fmt.Errorf("testRequest Variables is not valid json: %s", testRequest.Variables))
+	}
+	if !IsValidJson(testRequest.Response) {
+		panic(fmt.Errorf("testRequest Response is not json: %s", testRequest.Response))
+	}
+	return testRequest
+}
+
+func NewTestDataTemplater(templateDirs ...string) *TestDataTemplater {
+	var templateFiles []string
+	for _, dir := range []string{"./testdata/templates"} {
+		err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() {
+				templateFiles = append(templateFiles, path)
+			}
+			return nil
+		})
+		if err != nil {
+			panic(fmt.Errorf("error during loading template files: %s", err))
+		}
+	}
+
+	output := TestDataTemplater{}
+	tmpl, err := template.New("").Funcs(sprig.TxtFuncMap()).ParseFiles(templateFiles...)
+	if err != nil {
+		panic(fmt.Errorf("error during template initialization: %s", err))
+	}
+	output.rootTemplate = tmpl
+	return &output
+}
+
+type TestDataTemplater struct {
+	rootTemplate *template.Template
+}
+
+func (t *TestDataTemplater) ParseTemplatedString(contents string) string {
+	target, err := t.rootTemplate.Parse(contents)
+	if err != nil {
+		panic(fmt.Errorf("error parsing template: %s", err))
+	}
+	data := bytes.NewBuffer([]byte{})
+	if err = target.Execute(data, nil); err != nil {
+		panic(err)
+	}
+	return strings.TrimSpace(data.String())
+}
+
 type TestRequest struct {
 	Request   string
 	Variables string
 	Response  string
 }
 
+func (t *TestRequest) RequestWithVariables() string {
```

Comment:
@davidbloss can we log a follow up ticket to get rid of this.  I understand why its like this now but i'd love to kill this in a follow up.